### ISSUE TITLE
Cannot install supy using new OSGeo version (QGIS 3.20)

### ISSUE THREAD

**Issue by biglimp** [UMEP-dev/UMEP#310]
I cannot use either **pip install supy** outside as well as inside of a QGIS session. Error that comes up when try (I forced the installation to stop as it goes on trying to install other versions of **tables**:

```
C:\Users\xlinfr\Documents\PythonScripts\UMEP>pip install supy
Collecting supy
  Using cached supy-2021.5.26-py3-none-any.whl (400 kB)
Requirement already satisfied: scipy in c:\osgeo4w\apps\python39\lib\site-packages (from supy) (1.6.2)
Collecting numdifftools
  Using cached numdifftools-0.9.40-py2.py3-none-any.whl (99 kB)
Collecting supy-driver==2021a2
  Using cached supy_driver-2021a2-cp39-cp39-win_amd64.whl (2.0 MB)
Collecting f90nml
  Using cached f90nml-1.3.1.tar.gz (62 kB)
Collecting lmfit
  Downloading lmfit-1.0.2.tar.gz (306 kB)
     |████████████████████████████████| 306 kB 6.4 MB/s
Collecting atmosp
  Using cached atmosp-0.2.9-py3-none-any.whl (104 kB)
Collecting platypus-opt==1.0.4
  Using cached Platypus_Opt-1.0.4-py3-none-any.whl (70 kB)
Collecting dask
  Downloading dask-2021.6.2-py3-none-any.whl (973 kB)
     |████████████████████████████████| 973 kB 6.4 MB/s
Requirement already satisfied: matplotlib in c:\osgeo4w\apps\python39\lib\site-packages (from supy) (3.3.2)
Collecting multiprocess
  Downloading multiprocess-0.70.12.2-py39-none-any.whl (128 kB)
     |████████████████████████████████| 128 kB 6.4 MB/s
Requirement already satisfied: click in c:\osgeo4w\apps\python39\lib\site-packages (from supy) (8.0.1)
Collecting xarray
  Using cached xarray-0.18.2-py3-none-any.whl (807 kB)
Collecting cdsapi
  Using cached cdsapi-0.5.1.tar.gz (12 kB)
Collecting seaborn
  Using cached seaborn-0.11.1-py3-none-any.whl (285 kB)
Collecting tables
  Using cached tables-3.6.1.tar.gz (4.6 MB)
    ERROR: Command errored out with exit status 1:
     command: 'C:\OSGeo4W\bin\python3.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\xlinfr\\AppData\\Local\\Temp\\pip-install-o67ykg5j\\tables_504b0e1b54a74161a0d00c838cfa0d50\\setup.py'"'"'; __file__='"'"'C:\\Users\\xlinfr\\AppData\\Local\\Temp\\pip-install-o67ykg5j\\tables_504b0e1b54a74161a0d00c838cfa0d50\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\xlinfr\AppData\Local\Temp\pip-pip-egg-info-1b6kds_o'        
         cwd: C:\Users\xlinfr\AppData\Local\Temp\pip-install-o67ykg5j\tables_504b0e1b54a74161a0d00c838cfa0d50\
    Complete output (10 lines):
    H5closenggyiar6.c
    C:\Users\xlinfr\AppData\Local\Temp\H5closenggyiar6.c(2): warning C4013: 'H5close' undefined; assuming extern returning int
    LINK : fatal error LNK1181: cannot open input file 'hdf5.lib'
    * Using Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
    * USE_PKGCONFIG: False
    * Found HDF5 using system PATH ('C:\OSGeo4W\bin')
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/2b/32/847ee3f521aae6a0be380d923a736162d698586f444df1ac24b98c65025c/tables-3.6.1.tar.gz#sha256=49a972b8a7c27a8a173aeb05f67acb45fe608b64cd8e9fa667c0962a60b71b49 (from https://pypi.org/simple/tables/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check 
the logs for full command output.
  Using cached tables-3.5.2.tar.gz (7.8 MB)
    ERROR: Command errored out with exit status 1:
     command: 'C:\OSGeo4W\bin\python3.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\xlinfr\\AppData\\Local\\Temp\\pip-install-o67ykg5j\\tables_d95c7a61748e4498a736540f7dba9061\\setup.py'"'"'; __file__='"'"'C:\\Users\\xlinfr\\AppData\\Local\\Temp\\pip-install-o67ykg5j\\tables_d95c7a61748e4498a736540f7dba9061\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\xlinfr\AppData\Local\Temp\pip-pip-egg-info-xqxixj5b'        
         cwd: C:\Users\xlinfr\AppData\Local\Temp\pip-install-o67ykg5j\tables_d95c7a61748e4498a736540f7dba9061\
    Complete output (10 lines):
    H5closec8bafxw0.c
    C:\Users\xlinfr\AppData\Local\Temp\H5closec8bafxw0.c(2): warning C4013: 'H5close' undefined; assuming extern returning int
    LINK : fatal error LNK1181: cannot open input file 'hdf5.lib'
    * Using Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
    * USE_PKGCONFIG: False
    * Found HDF5 using system PATH ('C:\OSGeo4W\bin')
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/73/51/6dabb2b94826e5db3aa2542b80f1382780b96a0cd13e0cfb637b36ede5c5/tables-3.5.2.tar.gz#sha256=b220e32262bab320aa41d33125a7851ff898be97c0de30b456247508e2cc33c2 (from https://pypi.org/simple/tables/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached tables-3.5.1.tar.gz (8.3 MB)
ERROR: Operation cancelled by user
WARNING: You are using pip version 21.1.1; however, version 21.1.3 is available.
You should consider upgrading via the 'C:\OSGeo4W\bin\python3.exe -m pip install --upgrade pip' command.

C:\Users\xlinfr\Documents\PythonScripts\UMEP> 
```

**Comment by biglimp** [comment]
@sunt05, your pull request only worked halfway.

Actually, I found that it is not QGIS version that decides where the python is located, it is the actual python version. Hence, we should check for python version and if we have 3.9 should do as we do now, when we identify QGIS 3.20. If lower version of python we should use the old location of python.

I noticed this as I installed the long term version of QGIS (3.16) alongside 3.20. 3.16 is using the same python version (3.9.5)
